gobblin-core/src/main/java/org/apache/gobblin/converter/json/JsonSchema.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.json;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.converter.DataConversionException;
+import org.apache.gobblin.converter.avro.JsonElementConversionFactory.Type;
+import org.apache.gobblin.source.extractor.schema.Schema;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import static org.apache.gobblin.converter.avro.JsonElementConversionFactory.Type.ENUM;
+import static org.apache.gobblin.converter.avro.JsonElementConversionFactory.Type.FIXED;
+import static org.apache.gobblin.converter.avro.JsonElementConversionFactory.Type.RECORD;
+import static org.apache.gobblin.converter.avro.JsonElementConversionFactory.Type.UNION;
+import static org.apache.gobblin.converter.json.JsonSchema.SchemaType.CHILD;
+import static org.apache.gobblin.converter.json.JsonSchema.SchemaType.ROOT;
+
+
+/**
+ * Represents a source schema declared in the configuration with {@link ConfigurationKeys#SOURCE_SCHEMA}.
+ * The source schema is represented by a {@link JsonArray}.
+ * @author tilakpatidar
+ */
+public class JsonSchema extends Schema {
+  public static final String RECORD_FIELDS_KEY = "values";
+  public static final String TYPE_KEY = "type";
+  public static final String NAME_KEY = "name";
+  public static final String SIZE_KEY = "size";
+  public static final String ENUM_SYMBOLS_KEY = "symbols";
+  public static final String COLUMN_NAME_KEY = "columnName";
+  public static final String DATA_TYPE_KEY = "dataType";
+  public static final String COMMENT_KEY = "comment";
+  public static final String DEFAULT_VALUE_KEY = "defaultValue";
+  public static final String IS_NULLABLE_KEY = "isNullable";
+  public static final String DEFAULT_RECORD_COLUMN_NAME = "temp";
```

Comment:
What is DEFAULT_RECORD_COLUMN_NAME used for? If I understand correctly, it the column name for the base level of a JsonSchema. Maybe a name like "root" would be more meaningful?